jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/internal/SpanWrapper.java

```diff
@@ -0,0 +1,35 @@
+package io.opentracing.contrib.jaxrs.internal;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.opentracing.Span;
+
+/**
+ * Wrapper class used for exchanging span between filters.
+ *
+ * @author Pavol Loffay
+ */
+public class SpanWrapper {
+
+    private Span span;
+    private AtomicBoolean finished = new AtomicBoolean();
+
+    public SpanWrapper(Span span) {
+        this.span = span;
+    }
+
+    public Span span() {
+        return span;
+    }
+
+    public synchronized void finish() {
+        if (!finished.get()) {
+            finished.set(true);
+            span.finish();
```

Comment:
This [1] is for `MockSpan`, however from the discussion I assume it's better not to assume that `Span#finish()` is idempotent (unless it is defined in spec).

[1]: https://github.com/opentracing/opentracing-java/pull/72